tests/scripts/all.sh

```diff
@@ -2110,6 +2110,36 @@ component_test_psa_crypto_config_accel_hash_use_psa () {
     fi
 }
 
+component_test_psa_crypto_config_reference_hash_use_psa() {
+    msg "test: MBEDTLS_PSA_CRYPTO_CONFIG without accelerated hash and USE_PSA"
+    # start with full
+    scripts/config.py full
+    # use PSA config and disable driver-less algs as in the component
+    scripts/config.py set MBEDTLS_PSA_CRYPTO_CONFIG
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_STREAM_CIPHER
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_ECB_NO_PADDING
+    # disable options as in the component
+    # (no need to disable whole modules, we'll just skip their test suite)
+    scripts/config.py unset MBEDTLS_ECDSA_DETERMINISTIC
+    scripts/config.py -f include/psa/crypto_config.h unset PSA_WANT_ALG_DETERMINISTIC_ECDSA
+
+    make
+
+    msg "test: MBEDTLS_PSA_CRYPTO_CONFIG without accelerated hash and USE_PSA"
+    make test
+
+    # hidden option: when running outcome-analysis.sh, we can skip this
```

Comment:
Actually once this is done, we have not reason to run the "driver vs reference" part of `outcome-analysis.sh` manually, so we can remove this hidden option.

Also, we should run `ssl-opt.sh` in this reference component and make sure `analyze_outcomes.py` complains if there are any case skipped in the driver but not in reference (since `ssl-opt.sh` also reports in the outcome file, that's not harder that doing it for the test suites).

OTOH, `compat.sh` does not report to the outcome file, so there's no point running it here, since the only point of this component is for comparison.

Reply:
It looks like `compat.sh` reports to the outcome file. Results from my tests:
```
Linux-x86_64;component_test_psa_crypto_config_accel_hash_use_psa;tls13-compat;TLS 1.3 O->m: AES_128_GCM_SHA256,secp256r1,ecdsa_secp256r1_sha256;SKIP;
Linux-x86_64;component_test_psa_crypto_config_accel_hash_use_psa;tls13-compat;TLS 1.3 O->m: AES_128_GCM_SHA256,secp256r1,ecdsa_secp384r1_sha384;SKIP;
Linux-x86_64;component_test_psa_crypto_config_accel_hash_use_psa;tls13-compat;TLS 1.3 O->m: AES_128_GCM_SHA256,secp256r1,ecdsa_secp521r1_sha512;SKIP;
....
```

Reply:
I believe `tls13-compat` is a component of `ssl-opt.sh`, see `./tests/opt-testcases/tls13-compat.sh` and the way `opt-testcases/*.sh` is included from `ssl-opt.sh`.

Reply:
Righgt! I saw in results, that `compat` block is after `ssl-opt` and missed that `ls13-compat` is a component of `ssl-opt.sh`.

Reply:
Though in the long run, I think we should merge `compat.sh` into `ssl-opt.sh` for TLS 1.2 as well, the way it was done for 1.3. But that's a separate issue of course! (One of the benefits would be that we'd get support for outcome files for free, and more generally avoid duplicated scaffolding code between the two scripts.)

Reply:
`compat.sh` has a performance benefit though: it keeps the same server instance for multiple clients.